actions/ocr.go

```diff
@@ -1,293 +1,202 @@
-//				Id:    "",
-//				Data:  client.AdapterResult{Result: results[nodeIndex-1]},
-//				Error: nil,
-//			}
-//			nodesInitializer := client.HttpInitializer{
-//				Request:  client.HttpRequest{Path: fmt.Sprintf("/node_%d_contract_%d", nodeIndex, OCRInstanceIndex)},
-//				Response: client.HttpResponse{Body: adResp},
-//			}
-//			initializers = append(initializers, nodesInitializer)
-//		}
-//	}
-//
-//	err := i.Mockserver.PutExpectations(initializers)
-//	Expect(err).ShouldNot(HaveOccurred())
-//}
-//
-//// NewOCRSetupInputForObservability deploys and setups env and clients for testing observability
-//func NewOCRSetupInputForObservability(i *OCRSetupInputs, nodeCount int, contractCount int, rules map[string]*os.File) {
-//	By("Deploying environment",
-//		DeployOCRForEnv(
-//			i,
-//			environment.NewChainlinkClusterForObservabilityTesting(nodeCount),
-//		))
-//	By("Funding nodes", FundNodes(i))
-//	By("Deploying OCR contracts", DeployOCRContracts(i, contractCount))
-//
-//	expectations, err := GetMockserverInitializerDataForOTPE(
-//		i.OCRInstances,
-//		i.ChainlinkNodes,
-//	)
-//	Expect(err).ShouldNot(HaveOccurred())
-//	err = i.Mockserver.PutExpectations(expectations)
-//	Expect(err).ShouldNot(HaveOccurred())
-//
-//	err = i.SuiteSetup.Environment().DeploySpecs(environment.OtpeGroup())
-//	Expect(err).ShouldNot(HaveOccurred())
-//
-//	err = i.SuiteSetup.Environment().DeploySpecs(environment.PrometheusGroup(rules))
-//	Expect(err).ShouldNot(HaveOccurred())
-//}
-//
-//// NewOCRSetupInputForAtlas deploys and setups env and clients for testing atlas
-//func NewOCRSetupInputForAtlas(i *OCRSetupInputs, nodeCount int, contractCount int) {
-//	By("Deploying environment",
-//		DeployOCRForEnv(
-//			i,
-//			environment.NewChainlinkClusterForAtlasTesting(nodeCount),
-//		))
-//
-//	err := i.SuiteSetup.Environment().DeploySpecs(environment.AtlasEvmBlocksGroup())
-//	Expect(err).ShouldNot(HaveOccurred())
-//
-//	By("Funding nodes", FundNodes(i))
-//
-//	kafkaRestClient, err := environment.GetKafkaRestClientFromEnv(i.SuiteSetup.Environment())
-//	Expect(err).ShouldNot(HaveOccurred())
-//
-//	Eventually(func(g Gomega) []string {
-//		topics, err := kafkaRestClient.GetTopics()
-//		g.Expect(err).ShouldNot(HaveOccurred())
-//		return topics
-//	}, "3m", "1s").Should(ContainElements(
-//		ContainSubstring("block_headers"),
-//		ContainSubstring("transactions"),
-//	))
-//
-//	err = i.SuiteSetup.Environment().DeploySpecs(environment.AtlasEvmEventsAndReceiptsGroup())
-//	Expect(err).ShouldNot(HaveOccurred())
-//
-//	By("Deploying OCR contracts", DeployOCRContracts(i, contractCount))
-//}
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/smartcontractkit/helmenv/environment"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/contracts"
+	"math/big"
+	"time"
+)
+
+// OCRSetup contains the components needed for testing OCR
+type OCRSetup struct {
```

Comment:
I know this is from the older implementation, but I want to avoid needing custom structs for each action type, as it blocks composability of action functions between different types. It'd be much better to always pass the parameters in individually to the action functions.